### PR TITLE
Check if the cni binary is dir

### PR DESCRIPTION
**Pull request by niusmallnan**
https://github.com/rancher/rancher/issues/9324

### CODE REVIEW

File: binexec/watcher.go [hunk 1]
```diff
@@ -157,6 +157,14 @@ exec /usr/bin/nsenter -m -u -i -n -p -t %d -- $0 "$@"
 			break
 		}
 
+		fileInfo, err := os.Stat(p)
+		if err == nil && fileInfo.IsDir() {
+			if err = os.Remove(p); err != nil {
```
Comment (leodotcloud) [inline binexec/watcher.go]: could you please add a log here?

**Review by leodotcloud — CHANGES_REQUESTED** [review]

**Comment by leodotcloud** [comment]
@niusmallnan Can you please share your testing output for this issue?

- Start rancher/setup with your `network-manager` image.
- Exec into network-manager and do `rm -f /opt/cni/bin/rancher-bridge; mkdir -p /opt/cni/bin/rancher-bridge`
- Check if the log line is printed after metadata update?

**Comment by niusmallnan** [comment]
@leodotcloud 
```
time="2017-08-17T12:12:47Z" level=info msg="Setting up resolv.conf for ContainerId [031b276c73067e7a6d3622214c314b7a774017f4f03ac00d0508310dccbbac7f]"
time="2017-08-17T12:12:47Z" level=info msg="Setting up binaries for: map[rancher-bridge:031b276c73067e7a6d3622214c314b7a774017f4f03ac00d0508310dccbbac7f]"

time="2017-08-17T12:12:47Z" level=info msg="/opt/cni/bin/rancher-bridge is a dir, remove it"

time="2017-08-17T12:12:47Z" level=info msg="Setting up resolv.conf for ContainerId [a3b42a152788d3579a9a1225a4bab252fb55069c765884737013cebfa2aa8034]"
time="2017-08-17T12:12:47Z" level=info msg="CNI up" cid=a3b42a152788d3579a9a1225a4bab252fb55069c765884737013cebfa2aa8034 networkMode=ipsec
time="2017-08-17T12:12:47Z" level=info msg="CNI up done" cid=a3b42a152788d3579a9a1225a4bab252fb55069c765884737013cebfa2aa8034 networkMode=ipsec result=IP4:{IP:{IP:10.42.44.67 Mask:ffff0000} Gateway:10.42.0.1 Routes:[{Dst:{IP:169.254.169.250 Mask:ffffffff} GW:<nil>} {Dst:{IP:0.0.0.0 Mask:00000000} GW:10.42.0.1}]}, DNS:{Nameservers:[] Domain: Search:[] Options:[]}
time="2017-08-17T12:12:48Z" level=info msg="Applying new port rules"
```

**Comment by leodotcloud** [comment]
LGTM

**Review by leodotcloud — APPROVED** [review]

**Pull request merged**